service/handlers_keypairs.go

```diff
@@ -64,15 +63,34 @@ func KeypairListHandler(w http.ResponseWriter, r *http.Request) {
 	formatKeypairsResponse(true, "", "", "", keypairs, w)
 }
 
+func listAllowedKeypairs(authUser datastore.User) ([]datastore.Keypair, error) {
+	switch authUser.Role {
+	case 0:
+		fallthrough
```

Comment:
You mean the fallthrough?: That's jumping to the next subsequent case.
The default is set outside of the switch but I can change it to be the default switch case instead.
Now, one question for u:
As zero is not a good value to use directly for a role, I need a name for a const instead. Choose one:
- NoAuth
- Invalid
- ...?

And.. its value is good to be 0 or should I take it as -1? (That would involve explicit initialization for all User objects :S)